hydra/core/plugins.py

```diff
@@ -103,6 +105,63 @@ def is_in_toplevel_plugins_module(clazz: str) -> bool:
             "hydra._internal.core_plugins."
         )
 
+    @staticmethod
+    def _setup_plugin(
+        plugin: Any,
+        task_function: TaskFunction,
+        config: DictConfig,
+        config_loader: Optional[ConfigLoader] = None,
+        hydra_context: Optional[HydraContext] = None,
+    ) -> Any:
+        """
+        With HydraContext introduced in #1581, we need to set up the plugins in a way
+        that's compatible with both Hydra 1.0 and Hydra 1.1 syntax.
+        This method should be deleted in the next major release.
+        """
+        assert isinstance(plugin, Sweeper) or isinstance(plugin, Launcher)
+        assert (
+            config_loader is not None or hydra_context is not None
+        ), "config_loader and hydra_context cannot both be None"
+
+        param_keys = signature(plugin.setup).parameters.keys()
+
+        from hydra._internal.core_plugins.basic_launcher import BasicLauncher
+
+        if isinstance(plugin, BasicLauncher):
+            # BasicLauncher supports Hydra 1.0 style setup for compatibility with 3rd party Sweeper
+            plugin.setup(
+                config=config,
+                config_loader=config_loader,
+                task_function=task_function,
+                hydra_context=hydra_context,
+            )
```

Comment:
This is also unclear. Why should the Plugins interface handle the BasicSweeper in a special way?
Am I missing anything?

Reply:
removed. This is no longer needed.